org.adempiere.base/src/org/compiere/model/MAttachmentEntry.java

```diff
@@ -157,13 +170,13 @@ public String toString ()
 	public String toStringX ()
 	{
 		StringBuilder sb = new StringBuilder (m_name);
-		if (m_data != null)
+		if (getData() != null)
```

Comment:
this make toStringX() call rather expensive. perhaps not an issue if this is rarely use.

Reply:
Checked - is just used in two points in core:
MAttachment.addEntry when log level is FINE
WAttachment.displaySelected when log level is CONFIG

So, I think is safe.